xarray/core/missing.py

```diff
@@ -332,3 +369,162 @@ def _get_valid_fill_mask(arr, dim, limit):
+    -------
+    Interpolated Variable
+    """
+    if not indexes_coords:
+        return obj.copy()
+
+    # simple speed up for the local interpolation
+    if method in ['linear', 'nearest']:
+        obj, indexes_coords = _localize(obj, indexes_coords)
+
+    if method not in ['linear', 'nearest', 'zero', 'slinear', 'quadratic',
+                      'cubic', 'polynomial']:
+        raise ValueError('{} is not a valid method.'.format(method))
+
+    # default behavior
+    kwargs['bounds_error'] = kwargs.get('bounds_error', False)
+    # as coords should be always sorted, neglect this option
+    kwargs.pop('assume_sorted', None)
+
+    # target dimensions
+    dims = list(indexes_coords)
+    x, new_x = zip(*[indexes_coords[d] for d in dims])
+    destination = broadcast_variables(*new_x)
+
+    # transpose to make the interpolated axis to the last position
+    broadcast_dims = [d for d in obj.dims if d not in dims]
+    original_dims = broadcast_dims + dims
+    new_dims = broadcast_dims + list(destination[0].dims)
+    interped = interp_func(obj.transpose(*original_dims).data,
+                           x, destination, method, kwargs)
+
+    result = Variable(new_dims, interped, attrs=obj.attrs)
+
+    # transpose the result so that the new dimensions are inserted to the
+    # original position
+    if all(x1.dims == new_x1.dims for x1, new_x1 in zip(x, new_x)):
+        result_dims = obj.dims
+    else:
+        start = min(obj.get_axis_num(d) for d in dims)
+        result_dims = (broadcast_dims[:start] + list(destination[0].dims) +
+                       broadcast_dims[start:])
+    return result.transpose(*result_dims)
+
+
+def interp_func(obj, x, new_x, method, kwargs):
+    """
+    multi-dimensional interpolation for array-like.
+
+    Parameters
+    ----------
+    obj: np.ndarray or dask.array.Array
+        Array to be interpolated. The final dimension is interpolated.
+    x: a list of 1d array.
+        Original coordinates. Should not contain NaN.
+    new_x: 1d array
+        Original coordinates. Should not contain NaN.
+    method: string
+        {'linear', 'nearest', 'zero', 'slinear', 'quadratic', 'cubic'} for
+        1-dimensional itnterpolation.
+        {'linear', 'nearest'} for multidimensional interpolation
+    kwargs:
+        Optional keyword arguments to be passed to scipy.interpolator
+
+    Returns
+    -------
+    interpolated: array
+        Interpolated array
+
+    Note
+    ----
+    This requiers scipy installed.
+
+    See Also
+    --------
+    scipy.interpolate.interp1d
+    """
+    if not x:
+        return obj
+
+    if isinstance(obj, dask_array_type):
+        import dask.array as da
+
+        _assert_single_chunk(obj, range(obj.ndim - len(x), obj.ndim))
+        chunks = obj.chunks[:-len(x)] + new_x[0].shape
+        drop_axis = range(obj.ndim - len(x), obj.ndim)
+        new_axis = range(obj.ndim - len(x), obj.ndim - len(x) + new_x[0].ndim)
+        # call this function recursively
```

Comment:
I think the recursive use of this function could use a better explanation/comment.

Reply:
I removed this recursive call.